pandas/core/ops.py

```diff
@@ -754,7 +754,8 @@ def wrapper(self, other, axis=None):
         elif isinstance(other, pd.DataFrame):  # pragma: no cover
             return NotImplemented
         elif isinstance(other, (np.ndarray, pd.Index)):
-            if len(self) != len(other):
+            lo = len(np.atleast_1d(other))
+            if lo != 1 and len(self) != lo:
```

Comment:
no that's not a good thing to do, we don't like non-specific try:excepts.

you can specifically deal with it by something like:

``if lib.isscalar(lib.item_from_zerodim(other)):``